src/DataLib/fendl32B_retrofit/activation_analysis.py

```diff
@@ -0,0 +1,210 @@
+        the parent nuclide's KZA and the selected reaction type (MT).
+        This calculation determines both the residual nucleus, as described by the
+        daughter KZA value (dKZA) and the emitted particle(s).
+
+    Arguments:
+        MT (int): Unique identifier for the reaction type corresponding to a specific
+            reaction tabulated in the mt_table dictionary.
+        mt_dict (dict): Reference dictionary containing reaction information
+            for each MT number pre-defined in the ENDF manual.
+            (https://www.oecd-nea.org/dbdata/data/manual-endf/endf102_MT.pdf)
+        pKZA (int or str): Parent KZA identifier of the format ZZAAAM,
+            where ZZ is the isotope's atomic number, AAA is the mass number, 
+            and M is the isomeric state (0 if non-isomeric).
+    
+    Returns:
+        dKZA (str): KZA of the residual (daughter) nucleus.
+        emitted_particles (str): Name of the particles emitted from the reaction,
+            given as a single string. If multiple particles are emitted from the reaction,
+            then the emitted_particles would be written in the form "np", corresponding
+            to the emission of a neutorn and a proton.
+    """
+
+    try:
+        # Extract the parent nuclide properties from the pKZA
+        nucleus_protons = int(str(pKZA)[:2])
+        A = int(str(pKZA)[2:5])
+
+        # Identify the particles emitted from the given reaction
+        reaction = mt_dict[str(MT)]
+        emitted_particles = reaction.split(',')[1][:-1]
+        
+        # Tally the counts of each emitted particle from the reaction
+        particle_types = ['n', 'd', 'α', 'p', 't', '3He', 'gamma']
+        emission_tuples = [
+            (
+                count_emitted_particles(particle, emitted_particles),
+                particle
+            )
+            for particle in particle_types
+            if count_emitted_particles(particle, emitted_particles)
+        ]
+
+        # Reconfigure nucleus to account for changing nucleon counts
+        nucleus_neutrons, nucleus_protons = nuclear_decay(A,
+                                                          nucleus_protons,
+                                                          emission_tuples)
+        residual_A = str(nucleus_neutrons + nucleus_protons).zfill(3)
+        nucleus_protons = str(nucleus_protons).zfill(2)
+        M = isomer_check(emitted_particles)
+        if M != 0:
+            emitted_particles = emitted_particles[:-len(str(M))]
+        dKZA = f"{str(nucleus_protons)}{residual_A}{str(M)}"
+        return dKZA, emitted_particles
+    
+    except Exception as e:
+        logger.error(f"Error in reaction calculation for MT {MT}: {e}")
+        return None, None
+    
+def iterate_MTs(MTs, file_obj, mt_dict, pKZA):
+    # Initialize lists
+    cross_sections_by_MT = []
+    emitted_particles_list = []
+    dKZAs = []
+    groups = []
+
+    # Extract data for each MT
+    for MT in MTs:
+        try:
```

Comment:
Instead of using a `try` block, check beforehand to make sure that the MT is in the `mt_dict` and only proceed if `True`